coderd/rbac/authz.go

```diff
@@ -741,3 +745,45 @@ func rbacTraceAttributes(actor Subject, action policy.Action, objectType string,
 			attribute.String("object_type", objectType),
 		)...)
 }
+
+type authzCheckRecorderKey struct{}
+
+func WithAuthzCheckRecorder(ctx context.Context) context.Context {
+	return context.WithValue(ctx, authzCheckRecorderKey{}, ptr.Ref(AuthzCheckRecorder{
+		Checks: make(map[string]bool),
+	}))
+}
+
+type AuthzCheckRecorder struct {
+	// Checks is a map from preformatted authz check IDs to their authorization
+	// status (true => authorized, false => not authorized)
+	Checks map[string]bool
+}
+
+func recordAuthzCheck(ctx context.Context, check string, authorized bool) {
```

Comment:
I think you should pass the raw types into this, and let it handle serialization. It will also skip the cost of serializing the string when the check is not enabled.
```suggestion
func recordAuthzCheck(ctx context.Context, subject Subject, action policy.Action, object Object, authorized bool) {
```